src/GUI/gameloop.py

```diff
@@ -70,12 +94,154 @@ def setup(self, tmx_maps, player_start_pos):
+        hd_button_surface = pygame.Surface((200, 100))
+        hd_button_surface.fill('white')
+        third_button_surface = pygame.Surface((200, 100))
+        third_button_surface.fill('white')
+        back_button_surface = pygame.Surface((100, 100))
+        back_button_surface.fill('white')
+
+        FHD_BUTTON = Button(image=fhd_button_surface, pos=(623, 250), text_input="1920x1080", font=self.get_font(35), base_color='grey', hovering_color='black')
+        HD_BUTTON = Button(image=hd_button_surface, pos=(623, 365), text_input="1280x720", font=self.get_font(35), base_color='grey', hovering_color='black')
+        THIRD_BUTTON = Button(image=third_button_surface, pos=(623, 475), text_input="800x600", font=self.get_font(35), base_color='grey', hovering_color='black')
+        BACK_BUTTON = Button(image=back_button_surface, pos=(100, 650), text_input="< Back", 
+                                font=self.get_font(40), base_color='grey', hovering_color='black')
+
+        for button in [FHD_BUTTON, HD_BUTTON, THIRD_BUTTON, BACK_BUTTON]:
+            button.changeColor(RES_MOUSE_POS)
+            button.update(self.screen)
+
+        self.screen.blit(RES_TEXT, RES_RECT)
+
+        return BACK_BUTTON, FHD_BUTTON, HD_BUTTON, THIRD_BUTTON
+
+    def difficulty(self):
+        '''I made this option for now as a idea. Maybe we could implement a sort of difficulty system in the game?'''
+        self.screen.fill('white')
+
+        DIFF_MOUSE_POS = pygame.mouse.get_pos()
+
+        DIFF_TEXT = self.get_font(50).render('DIFFICULTY', True, 'black')
+        DIFF_RECT = DIFF_TEXT.get_rect(center=(623, 100))
+
+        easy_button_surface = pygame.Surface((200, 100))
+        easy_button_surface.fill('white')
+        medium_button_surface = pygame.Surface((200, 100))
+        medium_button_surface.fill('white')
+        hard_button_surface = pygame.Surface((200, 100))
+        hard_button_surface.fill('white')
+        back_button_surface = pygame.Surface((100, 100))
+        back_button_surface.fill('white')
+
+        EASY_BUTTON = Button(image=easy_button_surface, pos=(623, 250), text_input="Easy", font=self.get_font(35), base_color='grey', hovering_color='black')
+        MEDIUM_BUTTON = Button(image=medium_button_surface, pos=(623, 365), text_input="Medium", font=self.get_font(35), base_color='grey', hovering_color='black')
+        HARD_BUTTON = Button(image=hard_button_surface, pos=(623, 475), text_input="Hard", font=self.get_font(35), base_color='grey', hovering_color='black')
+        BACK_BUTTON = Button(image=back_button_surface, pos=(100, 650), text_input="< Back", font=self.get_font(40), base_color='grey', hovering_color='black')
+
+        for button in [EASY_BUTTON, MEDIUM_BUTTON, HARD_BUTTON, BACK_BUTTON]:
+            button.changeColor(DIFF_MOUSE_POS)
+            button.update(self.screen)
+
+        self.screen.blit(DIFF_TEXT, DIFF_RECT)
+
+        return BACK_BUTTON
+
+    def main_menu(self):
```

Comment:
I think main_menu should be in its own file. Like you did with button.py, to make the game loop from GUI less messy and just call main_menu from a variable.

Also, the constants you use in this method, are these used in other functions or code too or are they just for this method? Otherwise, I should not make them as constants but just normal variables.